### PR TITLE
Render lines using SpriteRenderer without breaking the batch.

### PR DESCRIPTION
**Pull request by pchote**
This PR removes the separate `color` shader, and instead handles it as a special case of the sprite renderer.  This avoids the shader switch when rendering projectiles and other contrails in RA and TD, saving several tens of ms render time (the difference between completely unplayable and kind of okay) during battles on a Raspberry Pi.

### CODE REVIEW

**Comment by reaperrr** [comment]
I've only tested this combined with #15148 so far, but even on a GTX 1070 the best-case gains on the RA shellmap reach around 4ms in heavy scenes (~9ms on pre-15148-bleed vs. 5ms with both PRs), and of particular notice is how the batches count stays almost completely flat at a low ~32-33 all the time vs. fluctuating wildly between ~60 and ~140 depending on the amount of on-screen contrails etc.
I'll hold off from approving for now until hopefully someone approves the code.

**Review by abcdefg30 — APPROVED** [review]

**Review by chrisforbes — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/639d468f7e646bb5d7aa9c668cfcd8df65923cff...aba88f7a31d30f9142ccdd6ad5be5d61fb115b20)

**Comment by reaperrr** [comment]
Out of curiosity, should this have a positive effect on laser beam render performance as well?
Because I don't see a (noticable) difference in my local testing.

**Pull request merged**
